crates/next-core/js/src/overlay/internal/container/Errors.tsx

```diff
@@ -166,41 +166,58 @@ function useDisplayState(
 }
 
 const enum TabId {
-  TurbopackIssues = "turbopack-issues",
+  TurbopackErrors = "turbopack-issues",
+  TurbopackWarnings = "turbopack-warnings",
   RuntimeErrors = "runtime-errors",
 }
 
+const TAB_PRIORITY = [
+  TabId.TurbopackErrors,
+  TabId.RuntimeErrors,
+  TabId.TurbopackWarnings,
+];
+
 export function Errors({ issues, errors }: ErrorsProps) {
   const [readyErrors, isLoading] = useResolvedErrors(errors);
 
-  const hasIssues = issues.length !== 0;
-  const hasIssueWithError = issues.some((issue) =>
+  const turbopackWarnings = issues.filter(
+    (issue) => !["bug", "fatal", "error"].includes(issue.severity)
+  );
+  const turbopackErrors = issues.filter((issue) =>
     ["bug", "fatal", "error"].includes(issue.severity)
   );
 
+  const hasTurbopackWarnings = turbopackWarnings.length > 0;
+  const hasTurbopackErrors = turbopackErrors.length > 0;
+
   const hasErrors = errors.length !== 0;
   const hasServerError = readyErrors.some((err) =>
     ["server", "edge-server"].includes(getErrorSource(err.error) || "")
   );
 
   // TODO for now it's already closable, but in future we might want to block users from using a broken app
-  // const isClosable = !isLoading && !hasIssueWithError && !hasServerError;
+  // const isClosable = !isLoading && !hasTurbopackErrors && !hasServerError;
   const isClosable = true;
 
   const defaultTab =
-    hasIssueWithError || !hasErrors
-      ? TabId.TurbopackIssues
-      : TabId.RuntimeErrors;
+    TAB_PRIORITY.filter(
+      (tabId) =>
+        ({
+          [TabId.TurbopackErrors]: turbopackErrors.length > 0,
+          [TabId.TurbopackWarnings]: turbopackWarnings.length > 0,
+          [TabId.RuntimeErrors]: hasErrors,
+        }[tabId])
+    )[0] ?? TabId.RuntimeErrors;
 
   const [selectedTab, setSelectedTab] = React.useState<string>(defaultTab);
 
   React.useEffect(() => {
-    if (defaultTab === TabId.TurbopackIssues) {
-      setSelectedTab(TabId.TurbopackIssues);
+    if (defaultTab === TabId.TurbopackErrors) {
+      setSelectedTab(TabId.TurbopackErrors);
```

Comment:
```suggestion
    if (defaultTab) {
      setSelectedTab(defaultTab );
```

Reply:
This is here to switch to the turbopack error tab if those appear later on, but I'm not sure if it's a good idea to have